pkg/sql/schemachanger/schemachanger_test: TestConcurrentSchemaChanges failed

pkg/sql/schemachanger/schemachanger_test.TestConcurrentSchemaChanges [failed](https://mesolite.cluster.engflow.com/invocations/default/cdc8464f-eeef-4930-b851-be31f7d07fd0?testReportRun=1&testReportShard=20&testReportAttempt=1#targets-Ly9wa2cvc3FsL3NjaGVtYWNoYW5nZXI6c2NoZW1hY2hhbmdlcl90ZXN0) on master @ [27468a22f82fb1e500bf7b35a62ca58d80abac39](https://github.com/cockroachdb/cockroach/commits/27468a22f82fb1e500bf7b35a62ca58d80abac39):


```
    schemachanger_test.go:759: RENAME TABLE TO t_149
    schemachanger_test.go:782: ADD COLUMN col_966 TO testdb_782.testsc_1.t_149
    schemachanger_test.go:708: RENAME DATABASE TO testdb_924
    schemachanger_test.go:708: RENAME DATABASE TO testdb_918
    schemachanger_test.go:782: ADD COLUMN col_7 TO testdb_782.testsc_1.t_149
    schemachanger_test.go:786: Parent database or schema or table is renamed or column already exists; skipping this column addition.
    schemachanger_test.go:708: RENAME DATABASE TO testdb_329
    schemachanger_test.go:810: Parent database or schema or table is renamed; skipping this column removal.
    schemachanger_test.go:806: DROP COLUMN col_966 FROM testdb_329.testsc_1.t_149
    schemachanger_test.go:733: DROP SCHEMA TO testsc_1
    schemachanger_test.go:782: ADD COLUMN col_144 TO testdb_329.testsc_1.t_149
    schemachanger_test.go:759: RENAME TABLE TO t_930
    schemachanger_test.go:759: RENAME TABLE TO t_694
    schemachanger_test.go:829: CREATE INDEX idx_827 ON testdb_329.testsc_1.t_694(col_144)
    schemachanger_test.go:782: ADD COLUMN col_290 TO testdb_329.testsc_1.t_694
    schemachanger_test.go:806: DROP COLUMN col_144 FROM testdb_329.testsc_1.t_694
    schemachanger_test.go:829: CREATE INDEX idx_99 ON testdb_329.testsc_1.t_694(col_290)
    schemachanger_test.go:782: ADD COLUMN col_360 TO testdb_329.testsc_1.t_694
    schemachanger_test.go:829: CREATE INDEX idx_74 ON testdb_329.testsc_1.t_694(col_290)
    schemachanger_test.go:829: CREATE INDEX idx_570 ON testdb_329.testsc_1.t_694(col_290)
    schemachanger_test.go:861: Parent database or schema or table is renamed or index is dropped; skipping this index removal.
    schemachanger_test.go:810: Parent database or schema or table is renamed; skipping this column removal.
    schemachanger_test.go:761: DROP TABLE t_725
    schemachanger_test.go:733: DROP SCHEMA TO testsc_1
    schemachanger_test.go:766: Parent database or schema is renamed; skipping this table renaming.
    schemachanger_test.go:766: Parent database or schema is renamed; skipping this table renaming.
    schemachanger_test.go:766: Parent database or schema is renamed; skipping this table renaming.
    schemachanger_test.go:766: Parent database or schema is renamed; skipping this table renaming.
    schemachanger_test.go:766: Parent database or schema is renamed; skipping this table renaming.
    schemachanger_test.go:786: Parent database or schema or table is renamed or column already exists; skipping this column addition.
    schemachanger_test.go:786: Parent database or schema or table is renamed or column already exists; skipping this column addition.
    schemachanger_test.go:786: Parent database or schema or table is renamed or column already exists; skipping this column addition.
    schemachanger_test.go:733: DROP SCHEMA TO testsc_1
    schemachanger_test.go:708: RENAME DATABASE TO testdb_930
    schemachanger_test.go:708: RENAME DATABASE TO testdb_101
    schemachanger_test.go:679: rename-db-worker encounters error pq: the new database name "testdb_329" already exists; signal to main routine and finish working
    schemachanger_test.go:675: drop-index-worker is signaled to finish work
    schemachanger_test.go:675: add-column-worker is signaled to finish work
    schemachanger_test.go:675: rename-tbl-worker is signaled to finish work
    schemachanger_test.go:675: drop-column-worker is signaled to finish work
    schemachanger_test.go:675: create-index-worker is signaled to finish work
    schemachanger_test.go:675: rename-schema-worker is signaled to finish work
    schemachanger_test.go:867: 
        	Error Trace:	pkg/sql/schemachanger/schemachanger_test_test/pkg/sql/schemachanger/schemachanger_test.go:867
        	Error:      	Received unexpected error:
        	            	pq: the new database name "testdb_329" already exists
        	Test:       	TestConcurrentSchemaChanges
    panic.go:626: -- test log scope end --
test logs left over in: outputs.zip/logTestConcurrentSchemaChanges397686955
--- FAIL: TestConcurrentSchemaChanges (105.09s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>run=1</code>
 - <code>shard=20</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/sql-foundations
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestConcurrentSchemaChanges.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>


Jira issue: CRDB-41162

pkg/sql/schemachanger/schemachanger_test.TestConcurrentSchemaChanges [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/16405011?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/16405011?buildTab=artifacts#/) on master @ [27468a22f82fb1e500bf7b35a62ca58d80abac39](https://github.com/cockroachdb/cockroach/commits/27468a22f82fb1e500bf7b35a62ca58d80abac39):


```
    schemachanger_test.go:761: DROP TABLE t_394
    schemachanger_test.go:759: RENAME TABLE TO t_670
    schemachanger_test.go:759: RENAME TABLE TO t_357
    schemachanger_test.go:782: ADD COLUMN col_809 TO testdb_794.testsc_866.t_357
    schemachanger_test.go:806: DROP COLUMN col_809 FROM testdb_794.testsc_866.t_357
    schemachanger_test.go:782: ADD COLUMN col_122 TO testdb_794.testsc_866.t_357
    schemachanger_test.go:761: DROP TABLE t_921
    schemachanger_test.go:786: Parent database or schema or table is renamed or column already exists; skipping this column addition.
    schemachanger_test.go:786: Parent database or schema or table is renamed or column already exists; skipping this column addition.
    schemachanger_test.go:786: Parent database or schema or table is renamed or column already exists; skipping this column addition.
    schemachanger_test.go:733: DROP SCHEMA TO testsc_866
    schemachanger_test.go:697: DROP DATABASE testdb_794
    schemachanger_test.go:782: ADD COLUMN col_423 TO testdb_794.testsc_866.t_357
    schemachanger_test.go:829: CREATE INDEX idx_651 ON testdb_794.testsc_866.t_357(col_423)
    schemachanger_test.go:861: Parent database or schema or table is renamed or index is dropped; skipping this index removal.
    schemachanger_test.go:810: Parent database or schema or table is renamed; skipping this column removal.
    schemachanger_test.go:786: Parent database or schema or table is renamed or column already exists; skipping this column addition.
    schemachanger_test.go:786: Parent database or schema or table is renamed or column already exists; skipping this column addition.
    schemachanger_test.go:786: Parent database or schema or table is renamed or column already exists; skipping this column addition.
    schemachanger_test.go:761: DROP TABLE t_80
    schemachanger_test.go:761: DROP TABLE t_946
    schemachanger_test.go:759: RENAME TABLE TO t_104
    schemachanger_test.go:731: RENAME SCHEMA TO testsc_483
    schemachanger_test.go:731: RENAME SCHEMA TO testsc_405
    schemachanger_test.go:731: RENAME SCHEMA TO testsc_742
    schemachanger_test.go:782: ADD COLUMN col_329 TO testdb_794.testsc_866.t_104
    schemachanger_test.go:733: DROP SCHEMA TO testsc_742
    schemachanger_test.go:708: RENAME DATABASE TO testdb_100
    schemachanger_test.go:782: ADD COLUMN col_700 TO testdb_100.testsc_742.t_104
    schemachanger_test.go:806: DROP COLUMN col_700 FROM testdb_100.testsc_742.t_104
    schemachanger_test.go:731: RENAME SCHEMA TO testsc_458
    schemachanger_test.go:731: RENAME SCHEMA TO testsc_399
    schemachanger_test.go:782: ADD COLUMN col_742 TO testdb_100.testsc_742.t_104
    schemachanger_test.go:766: Parent database or schema is renamed; skipping this table renaming.
    schemachanger_test.go:679: rename-db-worker encounters error pq: the new database name "testdb_671" already exists; signal to main routine and finish working
    schemachanger_test.go:675: drop-index-worker is signaled to finish work
    schemachanger_test.go:675: add-column-worker is signaled to finish work
    schemachanger_test.go:675: rename-schema-worker is signaled to finish work
    schemachanger_test.go:675: drop-column-worker is signaled to finish work
    schemachanger_test.go:675: create-index-worker is signaled to finish work
    schemachanger_test.go:761: DROP TABLE t_670
    schemachanger_test.go:675: rename-tbl-worker is signaled to finish work
    schemachanger_test.go:867: 
        	Error Trace:	pkg/sql/schemachanger/schemachanger_test_test/pkg/sql/schemachanger/schemachanger_test.go:867
        	Error:      	Received unexpected error:
        	            	pq: the new database name "testdb_671" already exists
        	Test:       	TestConcurrentSchemaChanges
    panic.go:626: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/ef3ec0efaf5a1e87629aae61874145ae/logTestConcurrentSchemaChanges3110935483
--- FAIL: TestConcurrentSchemaChanges (163.51s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestConcurrentSchemaChanges.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

pkg/sql/schemachanger/schemachanger_test.TestConcurrentSchemaChanges [failed](https://tanzanite.cluster.engflow.com/invocations/default/a488de16-6191-4596-88c4-e8aa799c25ea?testReportRun=4&testReportShard=20&testReportAttempt=1#targets-Ly9wa2cvc3FsL3NjaGVtYWNoYW5nZXI6c2NoZW1hY2hhbmdlcl90ZXN0) on master @ [27468a22f82fb1e500bf7b35a62ca58d80abac39](https://github.com/cockroachdb/cockroach/commits/27468a22f82fb1e500bf7b35a62ca58d80abac39):


```
    schemachanger_test.go:782: ADD COLUMN col_888 TO testdb_257.testsc_182.t_62
    schemachanger_test.go:782: ADD COLUMN col_628 TO testdb_257.testsc_182.t_62
    schemachanger_test.go:829: CREATE INDEX idx_518 ON testdb_257.testsc_182.t_62(col_888)
    schemachanger_test.go:829: CREATE INDEX idx_282 ON testdb_257.testsc_182.t_62(col_888)
    schemachanger_test.go:854: DROP INDEX idx_518 FROM testdb_257.testsc_182.t_62
    schemachanger_test.go:829: CREATE INDEX idx_864 ON testdb_257.testsc_182.t_62(col_888)
    schemachanger_test.go:708: RENAME DATABASE TO testdb_199
    schemachanger_test.go:708: RENAME DATABASE TO testdb_539
    schemachanger_test.go:829: CREATE INDEX idx_705 ON testdb_257.testsc_182.t_62(col_628)
    schemachanger_test.go:861: Parent database or schema or table is renamed or index is dropped; skipping this index removal.
    schemachanger_test.go:854: DROP INDEX idx_705 FROM testdb_539.testsc_182.t_62
    schemachanger_test.go:786: Parent database or schema or table is renamed or column already exists; skipping this column addition.
    schemachanger_test.go:782: ADD COLUMN col_285 TO testdb_539.testsc_182.t_62
    schemachanger_test.go:782: ADD COLUMN col_551 TO testdb_539.testsc_182.t_62
    schemachanger_test.go:829: CREATE INDEX idx_30 ON testdb_539.testsc_182.t_62(col_628)
    schemachanger_test.go:854: DROP INDEX idx_30 FROM testdb_539.testsc_182.t_62
    schemachanger_test.go:854: DROP INDEX idx_282 FROM testdb_539.testsc_182.t_62
    schemachanger_test.go:854: DROP INDEX idx_864 FROM testdb_539.testsc_182.t_62
    schemachanger_test.go:759: RENAME TABLE TO t_244
    schemachanger_test.go:759: RENAME TABLE TO t_266
    schemachanger_test.go:731: RENAME SCHEMA TO testsc_798
    schemachanger_test.go:810: Parent database or schema or table is renamed; skipping this column removal.
    schemachanger_test.go:782: ADD COLUMN col_953 TO testdb_539.testsc_798.t_266
    schemachanger_test.go:829: CREATE INDEX idx_302 ON testdb_539.testsc_798.t_266(col_888)
    schemachanger_test.go:829: CREATE INDEX idx_661 ON testdb_539.testsc_798.t_266(col_953)
    schemachanger_test.go:829: CREATE INDEX idx_525 ON testdb_539.testsc_798.t_266(col_285)
    schemachanger_test.go:829: CREATE INDEX idx_919 ON testdb_539.testsc_798.t_266(col_551)
    schemachanger_test.go:806: DROP COLUMN col_953 FROM testdb_539.testsc_798.t_266
    schemachanger_test.go:806: DROP COLUMN col_628 FROM testdb_539.testsc_798.t_266
    schemachanger_test.go:806: DROP COLUMN col_551 FROM testdb_539.testsc_798.t_266
    schemachanger_test.go:766: Parent database or schema is renamed; skipping this table renaming.
    schemachanger_test.go:786: Parent database or schema or table is renamed or column already exists; skipping this column addition.
    schemachanger_test.go:786: Parent database or schema or table is renamed or column already exists; skipping this column addition.
    schemachanger_test.go:679: create-index-worker encounters error pq: descriptor is being dropped; signal to main routine and finish working
    schemachanger_test.go:675: rename-schema-worker is signaled to finish work
    schemachanger_test.go:675: add-column-worker is signaled to finish work
    schemachanger_test.go:675: rename-tbl-worker is signaled to finish work
    schemachanger_test.go:675: drop-column-worker is signaled to finish work
    schemachanger_test.go:861: Parent database or schema or table is renamed or index is dropped; skipping this index removal.
    schemachanger_test.go:675: drop-index-worker is signaled to finish work
    schemachanger_test.go:697: DROP DATABASE testdb_539
    schemachanger_test.go:675: rename-db-worker is signaled to finish work
    schemachanger_test.go:867: 
        	Error Trace:	pkg/sql/schemachanger/schemachanger_test_test/pkg/sql/schemachanger/schemachanger_test.go:867
        	Error:      	Received unexpected error:
        	            	pq: descriptor is being dropped
        	Test:       	TestConcurrentSchemaChanges
    panic.go:626: -- test log scope end --
test logs left over in: outputs.zip/logTestConcurrentSchemaChanges2273071423
--- FAIL: TestConcurrentSchemaChanges (164.70s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>run=4</code>
 - <code>shard=20</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestConcurrentSchemaChanges.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

pkg/sql/schemachanger/schemachanger_test.TestConcurrentSchemaChanges [failed](https://tanzanite.cluster.engflow.com/invocations/default/4fa9c845-a5d6-447e-9011-6fc56d2450f9?testReportRun=25&testReportShard=20&testReportAttempt=1#targets-Ly9wa2cvc3FsL3NjaGVtYWNoYW5nZXI6c2NoZW1hY2hhbmdlcl90ZXN0) on master @ [451cdbe687b3303665bfd490af59c5df777c8909](https://github.com/cockroachdb/cockroach/commits/451cdbe687b3303665bfd490af59c5df777c8909):


```
    schemachanger_test.go:836: Parent database or schema or table is renamed or column is dropped or index already exists; skipping this index creation.
    schemachanger_test.go:733: DROP SCHEMA TO testsc_726
    schemachanger_test.go:782: ADD COLUMN col_839 TO testdb_983.testsc_726.t_24
    schemachanger_test.go:782: ADD COLUMN col_939 TO testdb_983.testsc_726.t_24
    schemachanger_test.go:786: Parent database or schema or table is renamed or column already exists; skipping this column addition.
    schemachanger_test.go:786: Parent database or schema or table is renamed or column already exists; skipping this column addition.
    schemachanger_test.go:786: Parent database or schema or table is renamed or column already exists; skipping this column addition.
    schemachanger_test.go:761: DROP TABLE t_4
    schemachanger_test.go:761: DROP TABLE t_897
    schemachanger_test.go:782: ADD COLUMN col_65 TO testdb_983.testsc_726.t_24
    schemachanger_test.go:782: ADD COLUMN col_164 TO testdb_983.testsc_726.t_24
    schemachanger_test.go:810: Parent database or schema or table is renamed; skipping this column removal.
    schemachanger_test.go:708: RENAME DATABASE TO testdb_156
    schemachanger_test.go:829: CREATE INDEX idx_269 ON testdb_983.testsc_726.t_24(col_65)
    schemachanger_test.go:829: CREATE INDEX idx_728 ON testdb_156.testsc_726.t_24(col_65)
    schemachanger_test.go:829: CREATE INDEX idx_824 ON testdb_156.testsc_726.t_24(col_164)
    schemachanger_test.go:733: DROP SCHEMA TO testsc_726
    schemachanger_test.go:786: Parent database or schema or table is renamed or column already exists; skipping this column addition.
    schemachanger_test.go:786: Parent database or schema or table is renamed or column already exists; skipping this column addition.
    schemachanger_test.go:786: Parent database or schema or table is renamed or column already exists; skipping this column addition.
    schemachanger_test.go:731: RENAME SCHEMA TO testsc_132
    schemachanger_test.go:731: RENAME SCHEMA TO testsc_768
    schemachanger_test.go:759: RENAME TABLE TO t_79
    schemachanger_test.go:761: DROP TABLE t_781
    schemachanger_test.go:782: ADD COLUMN col_404 TO testdb_156.testsc_768.t_79
    schemachanger_test.go:782: ADD COLUMN col_911 TO testdb_156.testsc_768.t_79
    schemachanger_test.go:829: CREATE INDEX idx_583 ON testdb_156.testsc_768.t_79(col_404)
    schemachanger_test.go:806: DROP COLUMN col_404 FROM testdb_156.testsc_768.t_79
    schemachanger_test.go:806: DROP COLUMN col_911 FROM testdb_156.testsc_768.t_79
    schemachanger_test.go:782: ADD COLUMN col_459 TO testdb_156.testsc_768.t_79
    schemachanger_test.go:829: CREATE INDEX idx_900 ON testdb_156.testsc_768.t_79(col_459)
    schemachanger_test.go:829: CREATE INDEX idx_423 ON testdb_156.testsc_768.t_79(col_459)
    schemachanger_test.go:829: CREATE INDEX idx_371 ON testdb_156.testsc_768.t_79(col_459)
    schemachanger_test.go:679: rename-db-worker encounters error pq: the new database name "testdb_860" already exists; signal to main routine and finish working
    schemachanger_test.go:675: rename-schema-worker is signaled to finish work
    schemachanger_test.go:675: create-index-worker is signaled to finish work
    schemachanger_test.go:675: drop-index-worker is signaled to finish work
    schemachanger_test.go:675: add-column-worker is signaled to finish work
    schemachanger_test.go:806: DROP COLUMN col_459 FROM testdb_156.testsc_768.t_79
    schemachanger_test.go:675: drop-column-worker is signaled to finish work
    schemachanger_test.go:759: RENAME TABLE TO t_751
    schemachanger_test.go:675: rename-tbl-worker is signaled to finish work
    schemachanger_test.go:867: 
        	Error Trace:	pkg/sql/schemachanger/schemachanger_test_test/pkg/sql/schemachanger/schemachanger_test.go:867
        	Error:      	Received unexpected error:
        	            	pq: the new database name "testdb_860" already exists
        	Test:       	TestConcurrentSchemaChanges
    panic.go:626: -- test log scope end --
test logs left over in: outputs.zip/logTestConcurrentSchemaChanges1561451893
--- FAIL: TestConcurrentSchemaChanges (139.49s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>run=25</code>
 - <code>shard=20</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestConcurrentSchemaChanges.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

pkg/sql/schemachanger/schemachanger_test.TestConcurrentSchemaChanges [failed](https://tanzanite.cluster.engflow.com/invocations/default/877cba2b-c531-482b-813f-e87ab3415f97?testReportRun=12&testReportShard=20&testReportAttempt=1#targets-Ly9wa2cvc3FsL3NjaGVtYWNoYW5nZXI6c2NoZW1hY2hhbmdlcl90ZXN0) on master @ [2f0b0399a5c9656bdf8a417d0ab57e4eb1ae0f64](https://github.com/cockroachdb/cockroach/commits/2f0b0399a5c9656bdf8a417d0ab57e4eb1ae0f64):


```
    schemachanger_test.go:829: CREATE INDEX idx_58 ON testdb_158.testsc_144.t_256(col_509)
    schemachanger_test.go:829: CREATE INDEX idx_690 ON testdb_158.testsc_144.t_256(col_509)
    schemachanger_test.go:854: DROP INDEX idx_690 FROM testdb_158.testsc_144.t_256
    schemachanger_test.go:782: ADD COLUMN col_632 TO testdb_158.testsc_144.t_256
    schemachanger_test.go:782: ADD COLUMN col_845 TO testdb_158.testsc_144.t_256
    schemachanger_test.go:782: ADD COLUMN col_764 TO testdb_158.testsc_144.t_256
    schemachanger_test.go:786: Parent database or schema or table is renamed or column already exists; skipping this column addition.
    schemachanger_test.go:786: Parent database or schema or table is renamed or column already exists; skipping this column addition.
    schemachanger_test.go:836: Parent database or schema or table is renamed or column is dropped or index already exists; skipping this index creation.
    schemachanger_test.go:733: DROP SCHEMA TO testsc_144
    schemachanger_test.go:697: DROP DATABASE testdb_158
    schemachanger_test.go:708: RENAME DATABASE TO testdb_825
    schemachanger_test.go:766: Parent database or schema is renamed; skipping this table renaming.
    schemachanger_test.go:759: RENAME TABLE TO t_568
    schemachanger_test.go:759: RENAME TABLE TO t_72
    schemachanger_test.go:759: RENAME TABLE TO t_447
    schemachanger_test.go:810: Parent database or schema or table is renamed; skipping this column removal.
    schemachanger_test.go:761: DROP TABLE t_826
    schemachanger_test.go:759: RENAME TABLE TO t_784
    schemachanger_test.go:861: Parent database or schema or table is renamed or index is dropped; skipping this index removal.
    schemachanger_test.go:761: DROP TABLE t_662
    schemachanger_test.go:761: DROP TABLE t_885
    schemachanger_test.go:782: ADD COLUMN col_759 TO testdb_825.testsc_144.t_784
    schemachanger_test.go:782: ADD COLUMN col_35 TO testdb_825.testsc_144.t_784
    schemachanger_test.go:782: ADD COLUMN col_982 TO testdb_825.testsc_144.t_784
    schemachanger_test.go:782: ADD COLUMN col_118 TO testdb_825.testsc_144.t_784
    schemachanger_test.go:782: ADD COLUMN col_692 TO testdb_825.testsc_144.t_784
    schemachanger_test.go:829: CREATE INDEX idx_181 ON testdb_825.testsc_144.t_784(col_759)
    schemachanger_test.go:806: DROP COLUMN col_759 FROM testdb_825.testsc_144.t_784
    schemachanger_test.go:782: ADD COLUMN col_603 TO testdb_825.testsc_144.t_784
    schemachanger_test.go:810: Parent database or schema or table is renamed; skipping this column removal.
    schemachanger_test.go:761: DROP TABLE t_839
    schemachanger_test.go:731: RENAME SCHEMA TO testsc_270
    schemachanger_test.go:731: RENAME SCHEMA TO testsc_400
    schemachanger_test.go:733: DROP SCHEMA TO testsc_400
    schemachanger_test.go:679: rename-schema-worker encounters error pq: schema "testsc_144" already exists; signal to main routine and finish working
    schemachanger_test.go:675: rename-db-worker is signaled to finish work
    schemachanger_test.go:675: drop-column-worker is signaled to finish work
    schemachanger_test.go:675: create-index-worker is signaled to finish work
    schemachanger_test.go:675: add-column-worker is signaled to finish work
    schemachanger_test.go:675: drop-index-worker is signaled to finish work
    schemachanger_test.go:675: rename-tbl-worker is signaled to finish work
    schemachanger_test.go:867: 
        	Error Trace:	pkg/sql/schemachanger/schemachanger_test_test/pkg/sql/schemachanger/schemachanger_test.go:867
        	Error:      	Received unexpected error:
        	            	pq: schema "testsc_144" already exists
        	Test:       	TestConcurrentSchemaChanges
    panic.go:626: -- test log scope end --
test logs left over in: outputs.zip/logTestConcurrentSchemaChanges1027498635
--- FAIL: TestConcurrentSchemaChanges (119.27s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>run=12</code>
 - <code>shard=20</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestConcurrentSchemaChanges.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

--- a/pkg/sql/schemachanger/schemachanger_test.go
+++ b/pkg/sql/schemachanger/schemachanger_test.go
@@ -684,6 +684,7 @@ func TestConcurrentSchemaChanges(t *testing.T) {
 		}
 	}
 
+	var nextObjectID atomic.Int64
 	// A goroutine that repeatedly renames database `testdb` randomly.
 	g.GoCtx(repeatWorkWithInterval("rename-db-worker", renameDBInterval, func(sqlDB *gosql.DB) error {
 		if err := useLegacyOrDeclarative(); err != nil {
@@ -697,7 +698,7 @@ func TestConcurrentSchemaChanges(t *testing.T) {
 			t.Logf("DROP DATABASE %v", dbName)
 			return createSchema()
 		}
-		newDBName := fmt.Sprintf("testdb_%v", rand.Intn(1000))
+		newDBName := fmt.Sprintf("testdb_%v", nextObjectID.Add(1))
 		if newDBName == dbName {
 			return nil
 		}
@@ -715,7 +716,7 @@ func TestConcurrentSchemaChanges(t *testing.T) {
 			return err
 		}
 		drop := rand.Intn(2) == 0
-		newSCName := fmt.Sprintf("testsc_%v", rand.Intn(1000))
+		newSCName := fmt.Sprintf("testsc_%v", nextObjectID.Add(1))
 		if scName == newSCName {
 			return nil
 		}
@@ -745,7 +746,7 @@ func TestConcurrentSchemaChanges(t *testing.T) {
 		if err := useLegacyOrDeclarative(); err != nil {
 			return err
 		}
-		newTblName := fmt.Sprintf("t_%v", rand.Intn(1000))
+		newTblName := fmt.Sprintf("t_%v", nextObjectID.Add(1))
 		drop := rand.Intn(2) == 0
 		var err error
 		if !drop {
@@ -774,7 +775,7 @@ func TestConcurrentSchemaChanges(t *testing.T) {
 			return err
 		}
 		dbName, scName, tblName := dbName, scName, tblName
-		newColName := fmt.Sprintf("col_%v", rand.Intn(1000))
+		newColName := fmt.Sprintf("col_%v", nextObjectID.Add(1))
 
 		_, err := sqlDB.Exec(fmt.Sprintf("ALTER TABLE %v.%v.%v ADD COLUMN %v INT DEFAULT %v",
 			dbName, scName, tblName, newColName, rand.Intn(100)))
@@ -814,7 +815,7 @@ func TestConcurrentSchemaChanges(t *testing.T) {
 
 	// A goroutine that creates secondary index on a randomly selected column.
 	g.GoCtx(repeatWorkWithInterval("create-index-worker", createIdxInterval, func(sqlDB *gosql.DB) error {
-		newIndexName := fmt.Sprintf("idx_%v", rand.Intn(1000))
+		newIndexName := fmt.Sprintf("idx_%v", nextObjectID.Add(1))
 
 		// Randomly pick a non-PK column to create an index on.
 		dbName, scName, tblName := dbName, scName, tblName
@@ -828,7 +829,8 @@ func TestConcurrentSchemaChanges(t *testing.T) {
 		if err == nil {
 			t.Logf("CREATE INDEX %v ON %v.%v.%v(%v)", newIndexName, dbName, scName, tblName, colName)
 		} else if isPQErrWithCode(err, pgcode.UndefinedDatabase, pgcode.UndefinedSchema,
-			pgcode.InvalidSchemaName, pgcode.UndefinedTable, pgcode.UndefinedColumn, pgcode.DuplicateRelation) {
+			pgcode.InvalidSchemaName, pgcode.UndefinedTable, pgcode.UndefinedColumn, pgcode.DuplicateRelation) ||
+			testutils.IsError(err, catalog.ErrDescriptorDropped.Error()) {
 			// Besides the potential name changes, it's possible this column has been
 			// dropped by the drop-column-worker or the secondary index name already
 			// exists.

